examples/llama/main.zig

```diff
@@ -1,156 +1,228 @@
+    \\--help                    print this help
+    \\--prompt <STRING>         the prompt
+    \\--model-config <PATH>     config.json path
+    \\--model-name <STRING>     model name
+    \\--model-weights <PATH>    model weights path
+    \\--model-tokenizer <PATH>  tokenizer path
+    \\--seed <UINT>             random seed (optional)
+    \\--seq-len <UINT>          sequence length
+    \\--create-options <STRING> platform creation options JSON, defaults to {}
+);
+
 pub fn main() !void {
     try asynk.AsyncThread.main(std.heap.c_allocator, asyncMain);
 }
 
 pub fn asyncMain() !void {
-    const CliArgs = struct {
-        pub const help =
-            \\ llama --model=llama3.7B.safetensors --tokenizer=vocab.json --num_layers=2
-        ;
-        model: []const u8,
-        tokenizer: ?[]const u8 = null,
-        layer_start: u8 = 0,
-        num_layers: ?u8 = null,
-        seq_len: u32 = 256,
-        topk: u32 = 2,
-        temperature: u32 = 1,
-        num_heads: ?i64 = null,
-        num_kv_heads: ?i64 = null,
-        rope_freq_base: ?i64 = null,
-        prompt: ?[]const u8 = null,
-        test_activations: ?[]const u8 = null,
-        seed: ?u128 = null,
-        // eg: --create-options='{"cuda":{"allocator":{"bfc":{"memory_fraction": 0.99}}}}'
-        create_options: []const u8 = "{}",
-    };
-
     log.info("   LLama was compiled with {}", .{@import("builtin").mode});
 
     const allocator = std.heap.c_allocator;
 
-    const tmp = try std.fs.openDirAbsolute("/tmp", .{});
-    try tmp.makePath("zml/llama/cache");
+    const parsers = comptime .{
+        .UINT = clap.parsers.int(usize, 0),
+        .STRING = clap.parsers.string,
+        .PATH = clap.parsers.string,
+    };
+    var diag: clap.Diagnostic = .{};
+    const stderr = std.io.getStdErr().writer();
+    var res = clap.parse(clap.Help, &params, parsers, .{
+        .diagnostic = &diag,
+        .allocator = allocator,
+    }) catch |err| {
+        diag.report(stderr, err) catch {};
+        stderr.print("usage: ", .{}) catch {};
+        clap.usage(stderr, clap.Help, &params) catch {};
+        stderr.print("\n", .{}) catch {};
+        return;
+    };
+    defer res.deinit();
+
+    if (res.args.help != 0) {
+        clap.help(std.io.getStdErr().writer(), clap.Help, &params, .{}) catch {};
+        return;
+    }
+
+    const config = blk: {
+        if (res.args.@"model-config") |config_json_path| {
+            var config_json_file = try asynk.File.open(config_json_path, .{ .mode = .read_only });
+            defer config_json_file.close() catch unreachable;
+            var reader = std.json.reader(allocator, config_json_file.reader());
+            defer reader.deinit();
+            const config_obj = try std.json.parseFromTokenSourceLeaky(llama.LlamaLM.Config, allocator, &reader, .{ .ignore_unknown_fields = true });
+            break :blk config_obj;
+        } else {
+            log.err("Missing --model-config", .{});
```

Comment:
clap doesn't support optional flags ?

Reply:
all flags are zig optionals in clap. wdym?

Reply:
I meant it doesn't support required flags